File: mobile-engage/src/main/java/com/emarsys/mobileengage/deeplink/DeepLinkAction.java
package com.emarsys.mobileengage.deeplink;

import android.app.Activity;

import com.emarsys.core.activity.ActivityLifecycleAction;
import com.emarsys.core.util.Assert;

public class DeepLinkAction implements ActivityLifecycleAction {

    private DeepLinkInternal deepLinkInternal;

    public DeepLinkAction(DeepLinkInternal deepLinkInternal) {
        Assert.notNull(deepLinkInternal, "DeepLinkInternal must not be null!");
        this.deepLinkInternal = deepLinkInternal;
    }

    @Override
    public void execute(Activity activity) {
        if (activity != null && activity.getIntent() != null) {
            deepLinkInternal.trackDeepLinkOpen(activity, activity.getIntent());
        }
    }

}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/deeplink/DeepLinkAction.kt
package com.emarsys.mobileengage.deeplink

import android.app.Activity
import com.emarsys.core.activity.ActivityLifecycleAction
import com.emarsys.core.activity.ActivityLifecyclePriorities
import com.emarsys.core.api.proxyApi
import com.emarsys.mobileengage.di.mobileEngage

class DeepLinkAction(private val deepLinkInternal: DeepLinkInternal,
                     override val priority: Int = ActivityLifecyclePriorities.DEEP_LINK_ACTION_PRIORITY,
                     override val repeatable: Boolean = true,
                     override val triggeringLifecycle: ActivityLifecycleAction.ActivityLifecycle = ActivityLifecycleAction.ActivityLifecycle.CREATE
) : ActivityLifecycleAction {

    override fun execute(activity: Activity?) {
        val intent = activity?.intent
        if (activity != null && intent != null) {
            deepLinkInternal.proxyApi(mobileEngage().concurrentHandlerHolder)
                .trackDeepLinkOpen(activity, intent, null)
        }
    }
}